src/OrleansRuntime/Hosting/SiloWrapper.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Orleans.Runtime.Hosting
+{
+    internal class SiloWrapper : ISilo
+    {
+        private readonly Silo silo;
+
+        public SiloWrapper(Silo silo, IServiceProvider services)
+        {
+            this.Services = services;
+            this.silo = silo;
+            this.Stopped = silo.SiloTerminated;
+        }
+
+        /// <inheritdoc />
+        public IServiceProvider Services { get; }
+
+        /// <inheritdoc />
+        public Task Stopped { get; }
+
+        /// <inheritdoc />
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            this.silo.Start();
+            return Task.CompletedTask;
```

Comment:
I think not, since that's very difficult to observe and test for. Cancellation must be occurring in parallel to the `StartAsync` call, so how can it distinguish between "about to return" and "just returned"?